src/DocumentDB.ChangeFeedProcessor.UnitTests/LeaseManagement/DocumentServiceLeaseTests.cs

```diff
@@ -102,12 +173,164 @@ public void ValidateSerialization_NullFields()
             var lease = (DocumentServiceLease)formatter.Deserialize(stream2);
 
             Assert.Null(lease.Id);
+            Assert.Null(lease.PartitionKey);
             Assert.Null(lease.ETag);
             Assert.Null(lease.PartitionId);
             Assert.Null(lease.Owner);
             Assert.Null(lease.ContinuationToken);
             Assert.Equal(new DocumentServiceLease().Timestamp, lease.Timestamp);
             Assert.Empty(lease.Properties);
         }
+
+        [Fact]
+        public void ValidateJsonSerialization_PartitionKey_NotFilled()
+        {
+
+            DocumentServiceLease originalLease = new DocumentServiceLease
+            {
+                Id = "id",
+                ETag = "etag",
+                PartitionId = "0",
+                Owner = "owner",
+                ContinuationToken = "continuation",
+                Timestamp = DateTime.Now - TimeSpan.FromSeconds(5),
+                Properties = new Dictionary<string, string> { { "key", "value" } }
+            };
+
+            var serializedLease = JsonConvert.SerializeObject(originalLease);
+            Assert.DoesNotContain("partitionKey", serializedLease);
+            var lease = JsonConvert.DeserializeObject<DocumentServiceLease>(serializedLease);
+            Assert.Null(lease.PartitionKey);
+        }
+
+        [Fact]
+        public void ValidateJsonSerialization_PartitionKey_Filled()
+        {
+
+            DocumentServiceLease originalLease = new DocumentServiceLease
+            {
+                Id = "id",
+                PartitionKey = "pk",
```

Comment:
Should this equal always to `id`? So this feature does not bring a different data distribution?

Reply:
Michael didnt want that behavior, he wanted for leases a new guid instead of reusing id value. since its a new collection there is no existing data distribution we have to worry about